### PR TITLE
KCRO-56: add ability to set and get i18n icon and screenshots for UI apps

### PR DESCRIPTION
**Pull request by icehess**

### CODE REVIEW

**Comment by icehess** [comment]
### Required PR:
* allow numbers in cb modules: https://github.com/2600hz/kazoo-ast/pull/11
* kzd_app accessors: https://github.com/2600hz/kazoo-core/pull/196
* https://github.com/2600hz/kazoo-core/pull/216

File: src/cb_apps_util.erl [hunk 1]
```diff
@@ -32,38 +41,105 @@ allowed_apps(AccountId) ->
...
+    end.
+
+-spec fetch_apps_whitelabel_overrides(AccountId::kz_term:ne_binary(), AppsMap::apps_map()) -> Overrides::apps_map().
+fetch_apps_whitelabel_overrides(AccountId, AppsMap) ->
+    AppIds = [app_whitelabel_doc_id(AppId) || AppId <- maps:keys(AppsMap)],
+    case kz_datamgr:open_docs(AccountId, AppIds) of
+        {'ok', JObjs} ->
+            maps:from_list(
+              [{kz_doc:id(JObj), kz_json:get_json_value(<<"doc">>, JObj)}
+               || JObj <- JObjs,
```
Comment (jamesaimonetti) [inline src/cb_apps_util.erl:120]: You can "cheat" by putting the get_json_value into another generator:
```erlang
[{kz_doc:id(JObj, Doc} || JObj <- JObjs, Doc <- [kz_json:get_json_value(<<"doc">>, JObj)], Doc =/= 'undefined']
```

File: src/cb_apps_util.erl [hunk 2]
```diff
@@ -235,57 +310,78 @@ get_whitelabel_doc(AccountId) ->
...
+            return_load_default(MasterAccountDb, [], ReturnAsMap)
     end.
 
--spec maybe_set_account(kz_term:ne_binary(), kz_json:object()) -> kz_json:object().
-maybe_set_account(Account, Doc) ->
+return_load_default(MasterAccountDb, JObjs, 'true') ->
+    %% next line we use Doc as default because this function may be called from outside
+    %% and app doc is directly opened
+    maps:from_list(
+      [{kz_doc:id(JObjs), ensure_master_account_id(MasterAccountDb, kz_json:get_value(<<"doc">>, JObj, JObj))}
```
Comment (jamesaimonetti) [inline src/cb_apps_util.erl]: Should be `kz_doc:id(JObj)` right? JObj not JObjs?

File: src/modules/cb_apps_store.erl [hunk 3]
```diff
@@ -268,6 +393,49 @@ delete(Context, _Id) ->
...
+delete(Context, _AppId, ?OVERRIDE) ->
+    crossbar_doc:delete(Context).
+
+-spec delete(cb_context:context(), path_token(), path_token(), path_token(), path_token()) -> cb_context:context().
+delete(Context, _AppId, ?OVERRIDE, ?SCREENSHOT, _Number) ->
+    AttachmentId = cb_context:fetch(Context, <<"attachment_id">>),
+    JObj = delete_image_from_doc(cb_context:doc(Context), AttachmentId, ?SCREENSHOT, get_lang(Context)),
+
+    %% removing attachment stub will remove attachment from document
+    cb_context:save(cb_context:set_doc(kz_doc:delete_attachment(JObj, AttachmentId))).
```
Comment (jamesaimonetti) [inline src/modules/cb_apps_store.erl]: `cb_context:save/1` doesn't exist?

Comment (jamesaimonetti) [reply src/modules/cb_apps_store.erl]: `cb_context:set_doc/1` doesn't exist either?

**Review by jamesaimonetti — CHANGES_REQUESTED** [review]

**Comment by icehess** [comment]
New requirement: https://github.com/2600hz/kazoo-core/pull/216

File: src/modules/cb_apps_store.erl (hunk 3 above)
Comment (icehess) [reply src/modules/cb_apps_store.erl]: fixed

File: src/cb_apps_util.erl (hunk 2 above)
Comment (icehess) [reply src/cb_apps_util.erl]: fixed

File: src/cb_apps_util.erl (hunk 1 above)
Comment (icehess) [reply src/cb_apps_util.erl:120]: I was trying to not cheat this time but anyway I changed it.

**Pull request merged**